plugins/repository-azure/src/test/java/org/elasticsearch/repositories/azure/AzureBlobStoreRepositoryTests.java

```diff
@@ -0,0 +1,190 @@
+
+                } else if (Regex.simpleMatch("GET /container/*", request)) {
+                    final BytesReference blob = blobs.get(exchange.getRequestURI().toString());
+                    if (blob == null) {
+                        exchange.sendResponseHeaders(RestStatus.NOT_FOUND.getStatus(), -1);
+                        return;
+                    }
+                    exchange.getResponseHeaders().add("Content-Type", "application/octet-stream");
+                    exchange.getResponseHeaders().add("x-ms-blob-content-length", String.valueOf(blob.length()));
+                    exchange.getResponseHeaders().add("x-ms-blob-type", "blockblob");
+                    exchange.sendResponseHeaders(RestStatus.OK.getStatus(), blob.length());
+                    blob.writeTo(exchange.getResponseBody());
+
+                } else if (Regex.simpleMatch("DELETE /container/*", request)) {
+                    Streams.readFully(exchange.getRequestBody());
+                    blobs.entrySet().removeIf(blob -> blob.getKey().startsWith(exchange.getRequestURI().toString()));
+                    exchange.sendResponseHeaders(RestStatus.ACCEPTED.getStatus(), -1);
+
+                } else if (Regex.simpleMatch("GET /container?restype=container&comp=list*", request)) {
+                    final Map<String, String> params = new HashMap<>();
+                    RestUtils.decodeQueryString(exchange.getRequestURI().getQuery(), 0, params);
+
+                    final StringBuilder list = new StringBuilder();
+                    list.append("<?xml version=\"1.0\" encoding=\"UTF-8\"?>");
+                    list.append("<EnumerationResults>");
+                    final String prefix = params.get("prefix");
+                    list.append("<Blobs>");
+                    for (Map.Entry<String, BytesReference> blob : blobs.entrySet()) {
+                        if (prefix == null || blob.getKey().startsWith("/container/" + prefix)) {
+                            list.append("<Blob><Name>").append(blob.getKey().replace("/container/", "")).append("</Name>");
+                            list.append("<Properties><Content-Length>").append(blob.getValue().length()).append("</Content-Length>");
+                            list.append("<BlobType>BlockBlob</BlobType></Properties></Blob>");
+                        }
+                    }
+                    list.append("</Blobs>");
+                    list.append("</EnumerationResults>");
+
+                    byte[] response = list.toString().getBytes(StandardCharsets.UTF_8);
+                    exchange.getResponseHeaders().add("Content-Type", "application/xml");
+                    exchange.sendResponseHeaders(RestStatus.OK.getStatus(), response.length);
+                    exchange.getResponseBody().write(response);
+
+                } else {
+                    exchange.sendResponseHeaders(RestStatus.INTERNAL_SERVER_ERROR.getStatus(), -1);
```

Comment:
This is a little strange isn't it? Azure doesn't actually return a `500` for all requests that go to non-existent paths right? :D

Reply:
I suppose Azure returns different errors depending on the type of operation and path. Since I want to keep this simple, I made it return 500 for any unsupported operations and/or non-existent path, like what has been done for S3 and GCS repository integration tests.

Reply:
True + I'm all for keeping it simple. But `500` seems wrong?
1. `5xx` behave differently form other requests in that they are potentially retried
2. I don't think Azure returns a `500` permanently for any route?

Maybe better go with a `404` or `400`?

Reply:
404 makes sense, you're right. I pushed 76bc3a04acc42e5a971c3256e454ca781e56da70

Reply:
404 makes sense, you're right. I pushed 76bc3a04acc42e5a971c3256e454ca781e56da70